Modules/JavaScript/Classes-Constructor-New/homework.js

```diff
@@ -0,0 +1,86 @@
+console.log(`
+  Task.
+
+    Create a Vector object that supports:
+     1. add (same length)
+     2. subtract
+     3. dot
+     4. norm
+     5. If you try to add, subtract, or dot two vectors with different lengths, you must throw an error.
+     6. toString
+     7. equals
+`);
+
+class Vector {
+  list = [];
+
+  constructor(list) {
+    this.list = list;
+  }
+
+  add(vector) {
+    this.checkIfVectorsLengthsMatch(vector);
+    
+    return new Vector(this.list.map((val, index) => val + vector.list[index]));
+  }
+  
+  subtract(vector) {
+    this.checkIfVectorsLengthsMatch(vector);
+    
+    return new Vector(this.list.map((val, index) => val - vector.list[index]));
+  }
+  
+  dot(vector) {
+    this.checkIfVectorsLengthsMatch(vector);
+    
+    return this.list.reduce((sum, val, index) => {
+      console.log('  ----- val=', val, 'vector.list[index]=', vector.list[index], 'addThis=', (val * vector.list[index]))
+      return sum + (val * vector.list[index]);
+    }, 0)
+  }
+  
+  norm() {
```

Comment:
this formula is incorrect

Reply:
yes, fixed

Reply:
This is still not fixed

Reply:
ahh, yes, updated only in console.log, now it should be fixed